src/transformStateWithClones.js

```diff
@@ -7,7 +7,45 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let addProperties = {};
+  let removeProperties = {};
+  let cleanProperties = {};
+  let addObject = state;
+
+  const array = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      addProperties = { ...addObject, ...action.extraData };
+      array.push(addProperties);
+      addObject = addProperties;
```

Comment:
You're directly assigning the modified state to 'addObject' which is then used in the next iteration. This can lead to unintended mutations of the state in subsequent actions. Instead, create a new object for each action.